module-apps/application-calllog/windows/CallLogDetailsWindow.cpp

```diff
@@ -207,6 +207,14 @@ namespace gui
     void CallLogDetailsWindow::onBeforeShow(ShowMode mode, SwitchData *data)
     {
         if (mode == ShowMode::GUI_SHOW_RETURN) {
+            if (auto contact = DBServiceAPI::MatchContactByPhoneNumber(application, record.phoneNumber);
```

Comment:
Please consider passing this info via switchData since this query can be quite heavy, depending on contacts number.

Reply:
When I return to this screen from editing a contact - SwitchData does not contain the new, updated data.
After consulting @PrzeBrudny , I made this update in the background with notification about the change of contact details in the database.